transport/nio/src/test/java/org/kaazing/gateway/transport/udp/UdpConnectorBM.java

```diff
@@ -0,0 +1,344 @@
+                ResourceAddress address = addressFactory.newResourceAddress(URI);
+                handler = new ConnectorHandler();
+                ConnectFuture future = udpConnector.connect(address, handler, null);
+                System.out.println("Connecting Udp connector client ..." + Thread.currentThread());
+                future.await(1, TimeUnit.SECONDS);
+                session = future.getSession();
+            } catch (Exception e) {
+                e.printStackTrace();
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class SocketState {
+        DatagramSocket udpClient;
+
+        public SocketState() {
+            try {
+                udpClient = new DatagramSocket();
+                udpClient.setSoTimeout(1000);
+                System.out.println("Connecting DatagramSocket client ..." + Thread.currentThread());
+                udpClient.connect(new InetSocketAddress("localhost", PORT));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @TearDown
+    public void destroy() throws Exception {
+        tcpAcceptor.dispose();
+        udpConnector.dispose();
+        schedulerProvider.shutdownNow();
+
+        udpServer.stop();
+
+        System.out.println();
+        System.out.println("Total client sent = " + clientSent + " client received = " + clientReceived);
+        System.out.println("Total server recv = " + serverReceived + "     server sent = " + serverSent);
+    }
+
+    // Aynchronous client using gateway's udp connector
+    // send, send, send ...
+    // receive, receive, receive ...
+    // It is possible that many messages are not received by server
+    @Benchmark
+    public void testAsyncConnector(ConnectorState state) throws Exception {
+        String message = "Hello World";
+        IoSessionEx session = (IoSessionEx) state.session;
+        ByteBuffer data = ByteBuffer.wrap(message.getBytes());
+        IoBufferAllocatorEx<?> allocator = session.getBufferAllocator();
+        WriteFuture writeFuture = session.write(allocator.wrap(data));
+        writeFuture.await(1, TimeUnit.SECONDS);
+        clientSent.incrementAndGet();
+    }
+
+    // Synchronous client using gateway's udp connector
+    // send - receive - send - receive ....
+    @Benchmark
+    public void testSyncConnector(ConnectorState state) throws Exception {
+        String message = "Hello World";
+        IoSessionEx session = (IoSessionEx) state.session;
+        ByteBuffer data = ByteBuffer.wrap(message.getBytes());
+        IoBufferAllocatorEx<?> allocator = session.getBufferAllocator();
+        CountDownLatch latch = new CountDownLatch(1);
+        state.handler.setLatch(latch);
+        session.write(allocator.wrap(data));
+        latch.await(1, TimeUnit.SECONDS);
+
+        clientSent.incrementAndGet();
+    }
+
+    // Synchronous client using DatagramSocket client
```

Comment:
I added a benchmark by running udp connector synchronously (after every sent message it waits for the echoed message before continuing further). IIRC, no kaazing client works slightly better.